src/site/content/en/blog/appcache-removal/index.md

```diff
@@ -0,0 +1,222 @@
+</div>
+
+### Why are tokens needed in multiple places?
+
+The **same origin trial token** needs to be associated with:
+
+- **All of your HTML pages** that use AppCache.
+- **All of your AppCache manifests** via the `ORIGIN-TRIAL` manifest field.
+
+If you've participated in origin trials in the past, you might have added the token just to your HTML pages. The AppCache "reverse" origin trial is special in that you need to associate a token with each of your AppCache manifests as well.
+
+Adding the origin trial token to your HTML pages enables the `window.applicationCache` interface from within your web apps. Pages that are not associated with a token won't be able to use `window.applicationCache` methods and events. Pages without a token also won't be able to load resources from the AppCache. Starting with Chrome 85, they will behave as if AppCache did not exist.
+
+Adding the origin trial token to your AppCache manifests indicates that each manifest is still valid. Starting with Chrome 85, any manifests that does not have an `ORIGIN-TRIAL` field will be treated as malformed, and the rules within the manifest will be ignored.
+
+### Origin trial deployment timing and logistics
+
+While the "reverse" origin trial officially starts with Chrome 84, you can [sign up](https://developers.chrome.com/origintrials/#/register_trial/1776670052997660673) for the origin trial today and add the tokens to your HTML and AppCache manifests. As your web app's audience gradually upgrades to Chrome 84, any tokens that you've already added will go into effect.
+
+Once you've added a token to your AppCache manifest, visit `chrome://appcache-internals` to confirm that your local instance of Chrome (version 84 or later) has properly associated the origin trial token with your manifest's cached entries. If your origin trial is recognized, you should see a field with `Token Expires: Tue Apr 06 2021...` on that page, associated with your manifest:
+
+<figure class="w-figure">
+  <img src="appcache-token.jpg" alt="chrome://appcache-internals interface showing a recognized token." width="550">
+</figure>
+
+## Testing prior to removal
+
+We strongly encourage you to migrate off of AppCache as soon as is feasible. If you want to test removal of AppCache on your web apps, use the `chrome://flags/#app-cache` [flag](https://www.chromium.org/developers/how-tos/run-chromium-with-flags) to simulate its removal. This flag is available starting with Chrome 84.
+
+## Migration strategies
+
+Service workers, which are widely supported in current browsers, offer an alternative to the offline experience provided by AppCache.
+
+We've provided a [polyfill](https://github.com/GoogleChromeLabs/sw-appcache-behavior) that uses a service worker to replicate some of the functionality of AppCache, though it does not replicate the entire AppCache interface. In particular, it does not provide a replacement for the `window.applicationCache` interface or the related AppCache events.
```

Comment:
For my own knowledge, whatever happened to that AppCache migration library that Workbox provided? I think this library was released before the "workbox" branding...

Reply:
It's the same library. We never release it as part of Workbox, just in the standalone project linked to in this paragraph.